nibabel/filebasedimages.py

```diff
@@ -347,7 +356,16 @@ def to_filename(self, filename):
         -------
         None
         '''
-        self.file_map = self.filespec_to_file_map(filename)
+        self.file_map = file_map = self.filespec_to_file_map(filename)
+        remove_env = os.environ.get('NIBABEL_REMOVE_BEFORE_TO_FILENAME', None)
+        for _, fh in file_map.items():
+            if not isinstance(fh, FileHolder):
+                continue
+            if os.path.lexists(fh.filename) and (remove_env or _is_annex_symlink(fh.filename)):
+                # Remove previous file where new file would be saved
+                # Necessary e.g. for cases where file is a symlink pointing
+                # to some non-writable file (e.g. under git annex control)
+                os.unlink(fh.filename)
         self.to_file_map()
```

Comment:
Yes, to do this properly would mean going down into `to_file_map`, because it's possible to save images with filenames that way, but then, at the moment, you'd have to go into the implementation for each file type (`to_file_map` not implemented by default).